app.py

```diff
@@ -131,10 +148,35 @@ def entry(entry_id):
     return render_template('pages/entry.html',  title="Entry" , entry=the_entry)
 
 
-@app.route('/add')
+@app.route('/add', methods=['GET', 'POST'])
 @login_required
 def new_entry():
-    return render_template('pages/new_entry.html',  title="New Entry")
+    form = NewEntryForm()
+    entries = mongo.db.entries
+    if form.validate_on_submit():
+
+        if form.image.name:
+            image = request.files[form.image.name]
+            uploaded_image = cloudinary.uploader.upload(image, width = 800, quality = 'auto')
+            image_url = uploaded_image.get('secure_url')
+        else:
+            image_url = ''
+        
+
+        tags = form.tags.data.split(',')
+        entries.insert({
+            "name": form.name.data,
+            "user_id": current_user.id,
+            "description": form.description.data,
+            "rating": int(form.rating.data),
+            "is_fav": form.is_fav.data,
+            "image" : image_url,
+            "tags": tags,
+            "created_on": datetime.now().strftime("%d/%m/%Y")
+        })
+        return redirect(url_for('listing'))
```

Comment:
That would be nice to display a successful message of some sort. Or maybe redirect to the page for the newly created entry?